packages/@lwc/engine-core/src/framework/modules/attrs.ts

```diff
@@ -18,36 +25,47 @@ export function patchAttributes(
     vnode: VBaseElement,
     renderer: RendererAPI
 ) {
-    const { attrs } = vnode.data;
+    const { attrs, external } = vnode.data;
     if (isUndefined(attrs)) {
         return;
     }
 
     const oldAttrs = isNull(oldVnode) ? EmptyObject : oldVnode.data.attrs;
+
+    // Attrs may be the same due to the static content optimization, so we can skip diffing
     if (oldAttrs === attrs) {
         return;
     }
```

Comment:
This static optimization was not in `attrs-unless-prop.ts`, so now we have the same optimization here